pkg/controller/neptune/dbcluster/setup.go

```diff
@@ -0,0 +1,232 @@
+	return nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.DeleteDBClusterInput) (bool, error) {
+	obj.DBClusterIdentifier = aws.String(meta.GetExternalName(cr))
+
+	if !*cr.Spec.ForProvider.DeletionProtection {
+		t := true
+		obj.SkipFinalSnapshot = &t
+	}
+	return false, nil
+}
+
+func lateInitialize(in *svcapitypes.DBClusterParameters, out *svcsdk.DescribeDBClustersOutput) error {
+	if out == nil || len(out.DBClusters) == 0 {
+		return nil
+	}
+
+	if in == nil {
+		in = &svcapitypes.DBClusterParameters{}
+	}
+
+	from := out.DBClusters[0]
+
+	in.AvailabilityZones = aws.LateInitializeStringPtrSlice(in.AvailabilityZones, from.AvailabilityZones)
+	in.BackupRetentionPeriod = aws.LateInitializeInt64Ptr(in.BackupRetentionPeriod, from.BackupRetentionPeriod)
+	in.CharacterSetName = aws.LateInitializeStringPtr(in.CharacterSetName, from.CharacterSetName)
+	in.DatabaseName = aws.LateInitializeStringPtr(in.DatabaseName, from.DatabaseName)
+	in.DBClusterParameterGroupName = aws.LateInitializeStringPtr(in.DBClusterParameterGroupName, from.DBClusterParameterGroup)
+	in.DBSubnetGroupName = aws.LateInitializeStringPtr(in.DBSubnetGroupName, from.DBSubnetGroup)
+	in.DeletionProtection = aws.LateInitializeBoolPtr(in.DeletionProtection, from.DeletionProtection)
+	in.EnableCloudwatchLogsExports = aws.LateInitializeStringPtrSlice(in.EnableCloudwatchLogsExports, from.EnabledCloudwatchLogsExports)
+	in.Engine = aws.LateInitializeStringPtr(in.Engine, from.Engine)
+	in.EngineVersion = aws.LateInitializeStringPtr(in.EngineVersion, from.EngineVersion)
+	in.EnableIAMDatabaseAuthentication = aws.LateInitializeBoolPtr(in.EnableIAMDatabaseAuthentication, from.IAMDatabaseAuthenticationEnabled)
+	in.KMSKeyID = aws.LateInitializeStringPtr(in.KMSKeyID, from.KmsKeyId)
+	in.MasterUsername = aws.LateInitializeStringPtr(in.MasterUsername, from.MasterUsername)
+	in.Port = aws.LateInitializeInt64Ptr(in.Port, from.Port)
+	in.PreferredBackupWindow = aws.LateInitializeStringPtr(in.PreferredBackupWindow, from.PreferredBackupWindow)
+	in.PreferredMaintenanceWindow = aws.LateInitializeStringPtr(in.PreferredMaintenanceWindow, from.PreferredMaintenanceWindow)
+	in.ReplicationSourceIdentifier = aws.LateInitializeStringPtr(in.ReplicationSourceIdentifier, from.ReplicationSourceIdentifier)
+	in.StorageEncrypted = aws.LateInitializeBoolPtr(in.StorageEncrypted, from.StorageEncrypted)
+
+	if len(in.VPCSecurityGroupIDs) == 0 && len(from.VpcSecurityGroups) != 0 {
+		in.VPCSecurityGroupIDs = make([]*string, len(from.VpcSecurityGroups))
+		for i, val := range from.VpcSecurityGroups {
+			in.VPCSecurityGroupIDs[i] = aws.LateInitializeStringPtr(in.VPCSecurityGroupIDs[i], val.VpcSecurityGroupId)
+		}
+	}
+	return nil
+}
+
+// nolint:gocyclo
+func isUpToDate(cr *svcapitypes.DBCluster, output *svcsdk.DescribeDBClustersOutput) (bool, error) {
+	in := cr.Spec.ForProvider
+	out := output.DBClusters[0]
+
+	if !cmp.Equal(in.BackupRetentionPeriod, out.BackupRetentionPeriod) {
```

Comment:
Most of these seem to be of basic types, such as `*string` or `*int64`. I'd suggest using `aws.StringValue` and friends and then do a basic equality check instead of `cmp.Equal`, which uses reflect library under the hood which is more costly.